### PR TITLE
week 6 assessment

### PR DESCRIPTION
**Pull request by jameshlee1**
# Submitting Week 6 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Coding Challenges
- [ ] Coding Concepts
- [ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+// Expected output: 1125
+
+// b) Create the function that makes the test pass.
+// Pseudo: Create a function called numbersArray that takes in an array of numbers as a parameter. Create a new variable  newNumbers then use the .map() method to take the index and cube each value. Return the variable newNumbers adding a + b
+
+const numbersArray = (array) => {
+  const newNumbers = array.map((index) => index**3)
+  return newNumbers.reduce((a, b) => a + b)
+}
+
```
Comment (triciasykes) [inline code-challenges.test.js:102]: nice use of reduce!  Your code is clean and efficient!  One way to level up would be to make some of them into oneliners, but not at all necessary!

File: code-challenges.test.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  { name: "arthur dent", occupation: "a radio employee" }
+]
+// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+
+// b) Create the function that makes the test pass.
+// Pseudo: Create a function called namesArrayCaps that takes in an array. Then use the .map() method to iterate through the objects and returning the value of each object. Using the split() method it turns into an array of strings. By using the .map() method and toUpperCase() method to capitalize each word and then join() them back into one string. Using string interpolation this will return the sentence and the name capitalized.
+
+const namesArrayCaps = (array) => {
+  return array.map((element) => element.name.split(" ").map((value) => value = (value[0].toUpperCase()) + value.slice(1)).join(" ") + ` is ${element.occupation}.`)
+}
```
Comment (DFWilliams) [inline code-challenges.test.js:37]: Nice one-liner!

**Review by DFWilliams — CHANGES_REQUESTED** [review]
Hey James! Looking through your assessment and you are missing your research answer for the first interview questions. Please add that and ping me on slack so I can finish reviewing your assessment. It's looking great btw!

File: code-challenges.test.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+const cubeAndSumTwo = [0, 5, 10]
+// Expected output: 1125
+
+// b) Create the function that makes the test pass.
+// Pseudo: Create a function called numbersArray that takes in an array of numbers as a parameter. Create a new variable  newNumbers then use the .map() method to take the index and cube each value. Return the variable newNumbers adding a + b
+
+const numbersArray = (array) => {
+  const newNumbers = array.map((index) => index**3)
+  return newNumbers.reduce((a, b) => a + b)
+}
```
Comment (DFWilliams) [inline code-challenges.test.js:101]: nice use of reduce! Your code is clean and efficient! One way to level up would be to make some of them into one-liners, but not at all necessary!

File: code-challenges.test.js (hunk 2 above)
Comment (DFWilliams) [inline code-challenges.test.js:37]: Nice one-liner!

File: interview-questions.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+First, without external resources. Challenge yourself to answer from memory.
+
+Then, research the question to expand on your answer. Even if you feel you have answered the question completely on your own, there is always something more to learn. Write your researched answer in your OWN WORDS.
+
+1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
+
+Your answer: the foreign key would be called cohort_id and added to the student table. Then you rails db:migrate to add a column to the table.
+
+Researched answer: a foreign key is a column in a database that are linked to a column in a different table. The foreign key column will be cohort_id and the foreign key Student.
+
```
Comment (DFWilliams) [inline interview-questions.md:14]: Both are great responses! I encourage you to add a little more content. Show off your knowledge, tell us what steps we would take to add that new column.

File: interview-questions.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+Researched answer: There is no RESTful route that always requires parameters to be passed. However, certain routes may require parameters based on the specific functionality that they provide. Routes used for searching, filtering, and routing to a specific instance typically require a params.
+
+3. Name three rails generator commands. What is created by each?
+
+Your answer: rails generate model - this will create a new model datatypes.
+rails generate controller - this will create a new controller that will give us to perform the CRUD actions on the model.
+rails generate migration - this creates a new migration of the original data which can later be modify or updated.
+
+Researched answer: rails resource - does everything Model command does, but also creates a controller, empty views folder, and resource routes., rails scaffold - it creates and completes the model, controller, and all the views needed for restful routing and full crud functionality.
```
Comment (DFWilliams) [inline interview-questions.md:27]: Awesome!

**Review by DFWilliams — APPROVED** [review]
Great work James! Your code is nice and clean and you're pseudocode looks good. You are definitely at a advance beginner level of comprehension. Keep pushing yourself, refactor, see if you can make your solutions a one-liner, and show off you knowledge in your interview questions. All and all you are doing awesome!

**Pull request merged**